pkgs/development/python-modules/pacup/default.nix

```
@@ -0,0 +1,48 @@
+{
+  lib,
+  poetry2nix,
+  python,
+  fetchFromGitHub,
+  fetchurl,
+}:
+poetry2nix.mkPoetryApplication rec {
+  pname = "pacup";
+  version = "1.1.0";
+
+  inherit python;
+
+  projectDir = ./.;
+  pyproject = fetchurl {
+    url = "https://github.com/pacstall/pacup/raw/${version}/pyproject.toml";
+    hash = "sha256-6fkJOnwDT+tDwIvtStYYQIPiz5hPUfIlSl6XNnI4YIc=";
+  };
+  poetrylock = fetchurl {
+    url = "https://github.com/pacstall/pacup/raw/${version}/poetry.lock";
+    hash = "sha256-/ZxwW30TE5vctN03Vjb8VPakTlBd1u88VrppVbmwH+Y=";
+  };
+
+  src = fetchFromGitHub {
+    owner = "pacstall";
+    repo = pname;
+    rev = version;
+    hash = "sha256-Hl/Gq/cZz4RGYKTuyDlrhATAUYEzKEuYIm0JdToN/ZY=";
+  };
```

I just tried this and it seems to work for me, is cleaner too.

```suggestion
  src = fetchFromGitHub {
    owner = "pacstall";
    repo = pname;
    rev = version;
    hash = "sha256-Hl/Gq/cZz4RGYKTuyDlrhATAUYEzKEuYIm0JdToN/ZY=";
  };
  projectDir = src;
```

---

And easier to maintain, thank you